docs/_advanced/authorization.md

```diff
@@ -0,0 +1,69 @@
+---
+title: Authorization
+lang: en
+slug: authorization
+order: 5
+---
+
+<div class="section-content">
+Authorization is the process of determining which Slack credentials should be available while processing an incoming Slack event.
+
+Apps installed on a single workspace can simply pass their bot token into the `App` constructor using the `token` parameter. However, if your app will be installed on multiple workspaces, you have two options. The easier option is to use the built-in OAuth support. This will handle setting up OAuth routes and verifying state. Read the section on [authenticating with OAuth](#authenticating-oauth) for details.
+
+For a more custom solution, you can set the `authorize` parameter to a function upon `App` instantiation. The `authorize` function should return [an instance of `AuthorizeResult`](https://github.com/slackapi/bolt-python/blob/main/slack_bolt/authorization/authorization_result.py#L6), which contains information about who and where the event is coming from.
+
+`AuthorizeResult` should have a few specific properties, all of type `str`:
+- Either **`bot_token`** (xoxb) *or* **`user_token`** (xoxp) are **required**. Most apps will use `bot_token` by default. Passing a token allows built-in functions (like `say()`) to work.
+- **`bot_user_id`** and **`bot_id`**, if using a `bot_token`. You can fetch this information by using the `from_auth_test_response()` function, which is a method on `AuthorizeResult`.
+- **`enterprise_id`** and **`team_id`**, which can be found in many events that are sent to your app.
+- **`user_id`** only when using `user_token`.
+</div>
+
+```python
```

Comment:
This is the only sample I haven't run locally because I couldn't get the version with the authorize function added. I'm guessing there's some mistakes. I will test this code sample and fix it in the near future.